src/components/smartSearch/filters/PersonTags/DisplayPersonTags.tsx

```diff
@@ -18,7 +19,14 @@ const DisplayPersonTags = ({ filter }: DisplayPersonTagProps) : JSX.Element => {
     const tagsQuery = useQuery(['tags', orgId], getTags(orgId as string));
     const tags = tagsQuery?.data || [];
 
-    const selectedTags = tags.filter(t => tagIds.includes(t.id));
+    // preserve the order of the tag array
+    const selectedTags = tagIds.reduce((acc: ZetkinTag[], id) => {
```

Comment:
Why use `reduce()` for what is essentially just a `filter()`? This is unnecessarily unoptimized as it will generate a new array over and over again, and at the same time is more difficult to read.

Reply:
Oh I see now what this is doing. It's converting the list of `tagIds` (IDs only) to tag objects. I would use a `map()` for that:

```tsx
const selectedTags = tagIds.map(id => tags.find(tag => tag.id === id));
```

Reply:
I did!
And then I had to filter it out to exclude potential `undefined`s, so Typescript wouldn't complain, but it STILL didn't interpret correctly after the filter for whatever reason, so @djbusstop suggested I use reduce.
Better solution?

Reply:
Oh, I see. Well, I don't like changing logic to be less readable just to satisfy the type-checker. There are other ways of doing that (e.g. `as`). But you can add that to the refactor list instead and move ahead with the merge if you want.

Reply:
added to refactor list, but just for my understanding -- 
it's possible that the map would return undefined if a tag was deleted, right? 
would the desired behaviour be then to actually _want_ the selectedTags array to be of type (ZetkinTag | undefined)[], so that the autocomplete actually takes that as the options and conditionally renders something for deleted tags?

Reply:
Yeah, that's a better idea!